File: Lista 5/Python/Ponto2D.py
from math import sqrt

class Ponto2D:
    def __new__(cls, x, y):
        obj = super().__new__(cls)
        obj.__x = x
        obj.__y = y
        return obj

    def __init__(self, x, y):
        pass

    def __str__(self):
        return f"({self.__x}, {self.__y})"
    
    @property
    def x(self):
        return self.__x
    
    @property
    def y(self):
        return self.__y
    
    def isEixoX(self):
        return self.__x == 0
    
    def isEixoY(self):
        return self.__y == 0

    def isEixos(self):
        return self.__x == 0 or self.__y == 0

    def quadrant(self):
        if self.isEixos():
            return 0
        
        if self.__x > 0:
            if self.__y > 0:
                return 1
            else:
                return 4
        else:
            if self.__y > 0:
                return 2
            else:
                return 3
            
    def distance(self, Ponto2D):
        distancia = sqrt((Ponto2D.x - self.__x)**2 + (Ponto2D.y - self.__y)**2)
        return distancia


if __name__ == "__main__":
    ponto = Ponto2D(1, 0)
    print(ponto)
    print("IsEixoX:", ponto.isEixoX())
    print("IsEixoY:", ponto.isEixoY())
    print("IsEixos:", ponto.isEixos())

    ponto2 = Ponto2D(4, 1)
    print("Quadrante:", ponto2.quadrant())
    print(ponto.distance(ponto2))

File: Lista 5/Java/Ponto2D.java
import java.lang.Math;

public class Ponto2D {
    private final double x;
    private final double y;

    public Ponto2D(double x, double y) {
        this.x = x;
        this.y = y;
    }
    
    public void imprimirPonto() {
        System.out.println("(" + x + ", " + y + ")");
    }

    public boolean isEixoX() {
        return x == 0;
    }

    public boolean isEixoY() {
        return y == 0;
    }

    public boolean isEixos() {
        return (x == 0 || y == 0);
    }

    public int quadrante() {
        if (x == 0 || y == 0)
            return 0;
        
        if (x > 0)
            return ((y > 0) ? 1 : 4);
        else
            return ((y > 0) ? 2 : 3);
    }

    // Getters
    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public double distancia(Ponto2D ponto) {
        double distancia = Math.sqrt(((ponto.getX() - x) * (ponto.getX() - x)) + ((ponto.getY() - y) * (ponto.getY() - y)));
        return distancia;
    }

    
    public static void main(String[] args){
        Ponto2D ponto1 = new Ponto2D(4, 0);
        Ponto2D ponto2 = new Ponto2D(0, 3);

        ponto1.imprimirPonto();
        System.out.println(ponto1.isEixoX());
        System.out.println(ponto1.isEixoY());
        System.out.println(ponto1.isEixos());

        System.out.println(ponto1.distancia(ponto2));
    }
}
